packages/sdk/src/NomadContext.ts

```diff
@@ -173,6 +180,51 @@ export class NomadContext extends MultiProvider<config.Domain> {
     return this.mustGetCore(this.governor.domain);
   }
 
+  /**
+   * Proves and Processes a transaction on the destination chain. This is subsidize and
+   * automatic on non-Ethereum destinations
+   * 
+   * @dev Ensure that a transaction is ready to be processed. You should ensure the following
+   * criteria have been met prior to calling this function:
+   *  1. the tx has been relayed
+   *  2. the `confirmAt` timestamp for the tx is in the past
+   *  3. the user is on the destination chain in their wallet
+   *
+   * @returns The Contract Transaction receipt
+   */
+  async process(origin: string | number, txId: string): Promise<ContractTransaction>{
+    // get s3 proof URL for environment
+    const { environment } = this.conf
+    const s3URL = s3URLs[environment]
+
+    const message = (await NomadMessage.baseFromTransactionHash(
```

Comment:
hmmmm maybe we add some version of `singleFromTransactionHash` that doesn't throw if there are multiple messages? `firstFromTransactionHash`? hmmmmmmmmmmmmmmmmmmm

actually maybe even better, this function takes in a `NomadMessage` instead of an `origin` and `txId`?

Reply:
or make single not throw?